packages/core/src/shared/filetypes.ts

```diff
@@ -362,7 +361,7 @@ export const codefileExtensions = new Set([
 ])
 
 // Code file names without an extension
-export const codefileNames = new Set(['Dockerfile', 'Dockerfile.build', 'gradlew', 'mvnw'])
+export const codefileNames = new Set(['Dockerfile', 'Dockerfile.build', 'gradlew', 'mvnw', '.gitignore'])
 
 // Build file names
 export const buildfileNames = new Set(['gradle/wrapper/gradle-wrapper.jar'])
```

Comment:
This should be in `codefileNames`. It's literally being treated as a codefile, so the separate variable is just confusing.

Reply:
missed in my rebase. we actually don't need to particularly check this jar file since we need any other files included in repository when we want to build the code (e.g. icon/png files in the TS repos). so removed `buildfileNames`